src/Token/TokenProvider.php

```diff
@@ -92,11 +90,11 @@ public static function defaultProvider(array $config = [])
      *
      * @return callable
      */
-    public static function fromToken(TokenInterface $token)
+    public static function fromToken(TokenInterface $token): callable
```

Comment:
I think we should avoid adding type hints to existent functions. I also made a similar comment above.